### PR TITLE
performance improvement

### PR DESCRIPTION
**Pull request by thilo**
access properties through a hash instead of a list.

### CODE REVIEW

File: lib/couch_potato/persistence/simple_property.rb [hunk 1]
```diff
@@ -33,18 +33,18 @@ def build(object, json)
       end
 
       def dirty?(object)
-        object.send("#{name}_changed?")
+        object.send("#{name}_changed?".freeze)
       end
 
       def serialize(json, object)
-        json[name] = @type_caster.cast_back object.send(name)
```
Comment (langalex) [inline lib/couch_potato/persistence/simple_property.rb]: why did you remove this?

File: lib/couch_potato/persistence/properties.rb [hunk 2]
```diff
@@ -11,30 +11,42 @@ class PropertyList
...
-          @list << property
+          list[property.name] = property
+        end
+        
+        def []=(key, property)
+          list[key] = property
         end
 
+        def [](key)
+          (@all_list ||= inherited_properties.merge(list))[key]
```
Comment (langalex) [inline lib/couch_potato/persistence/properties.rb]: duplicate code with line 18. also not sure if caching this could lead to problems. in general the performance benefit should be minimal anyway since properties are only initialized once on app startup.

**Comment by langalex** [comment]
updated master today (merged commits from 1.x branch) and looks like this or something similar was already implemented in the 1.x branch.

**Comment by thilo** [comment]
Yes totally possible. I will remove my branch for that.
